accepted/2024/[TS]PMUI-Readme-rendering.md

```diff
@@ -0,0 +1,216 @@
+# Markdown README Rendering in PM UI
+
+- Jonatan Gonzalez ([jgonz120](https://github.com/jgonz120)) 
+- Issue [#12583](https://github.com/NuGet/Home/issues/12583) <!-- GitHub Issue link -->
+- [Feature Spec](https://github.com/NuGet/Home/blob/7943122dffa435f4daeee600efcc5b744cd2e97e/accepted/2023/PMUI-README-rendering.md)
+
+## Summary
+
+We want to update the packages pane in the PMUI to render the README for the selected package. 
+We also need to update the Nuget Server API to provide a link to download the README directly, without having to download the nupkg first.
+
+## Motivation 
+
+The README file is an essential source of information that can help customers understand what a package does.
+By rendering the README file directly in the PM UI, customers can easily access the information they need to determine if a package will be helpful to them while searching, browsing, and managing their projects’ packages.
+This enhancement streamlines the user experience and may encourage package authors to create comprehensive README documentation when customers will be more likely to rely on it for useful information about the package.
+
+## Explanation
+### Functional explanation
+#### PM UI
+![Alt text](https://github.com/NuGet/Home/assets/89422562/81b24877-f12f-4783-905c-4a155d3c7693)
+
+
+The PM UI will be updated to have tabs for the Package Details and the the README.
+This UX will be displayed for both the Browse and installed tabs.
+It will also be displayed for both the solution level and project level package managers.
+
+##### README Tab
+This tab is rendered if a remote source available to download the readme or the Installed tab is selected and the selected version is found in the global packages folder.
+This is to ensure that users using the Browse tab will not see the README tab if they do not have a remote source that allows README downloads.
+
+If the README tab is rendered and we do not find a README then a message is rendered encouraging the user to contact the package author to upload a README.
+![alt text](../../meta/resources/ReadMePMUI/NoReadMeFound.png)
+
+##### Package Details
+This tab is always rendered and contains the package details information along with the Vulnerabilty and Depreciation information. 
```

Comment:
Should the vulnerability warning area be moved above the tabs so that it's always visible? 

I'm thinking that it's not good to "hide" the warning down in the tab, so that customers who use the README tab (especially if they use it primarily, or exclusively) will never see this vulnerability info:

![image](https://github.com/user-attachments/assets/64dd5af8-833c-43d6-bcb3-7b24881d44ec)

Reply:
Moving it outside would make the readme difficult to read. Some options that come to mind

1. Add the warning icon to the Package Details Tab when there is a Vulnerability/Deprecation. 
2. Move both of these outside and make them collapsable. 
3. Default to the Package Details tab if there are any vulnerabilities or deprecations.
4. Move Vulnerabilities and Deprecations to their own tabs.

I think the 3rd option would be the easiest option.

Reply:
I prefer 1&2, maybe even 4. 
3 Is my least preference as it essentially stomps on the user's choice for default tab (or last used tab).

Reply:
This is a great point about keeping vulnerability info visible.
@jgonz120 those are some good options to handle the scenario. If a vulnerability is present, I could see that as justifiable reason for showing the details tab with the vulnerability info at the top. If we were to consider option 4, it should also default to the tab w/ vulnerability info if there is a vulnerability present. For good measure, let's run these options by the UX board.

Reply:
Is it okay to resolve this as I follow up with the UI board? This feels more like an implementation detail, I can change the screenshot to not include the vulnerability information if that helps.

Reply:
I feel it's also a design detail. When we show informative text around vulnerabilities, it's now moving from a permanent display to a tabular display. That said, yes, I believe it's acceptable to address it in a follow-up PR based on UX board feedback.